### PR TITLE
Site template selection

### LINKED ISSUE #9: Theme Selection

### LINKED ISSUE #266: Selecting a Template List via ID
Right now a template/theme is selected via its "Name".
I've changed the backend to select accept the template via database ID.

### PR DESCRIPTION
**Pull request by ArsalanSavand**
Fixes #266 
Fixes #9

### CODE REVIEW

File: app/api/api_service.js [hunk 1]
```diff
@@ -79,6 +79,12 @@ function API($http, $resource, ENV) {
       name: "SetSiteTemplateConfig",
       endpoint: "website/site/:site_id/set-template-config/" // PUT
     }, {
+      name: "SiteTemplates",
+      endpoint: "website/site/:site_id/templates/" // GET
```
Comment (Alir3z4) [inline app/api/api_service.js]: This should be `siteId`

File: app/site/site_controller.js [hunk 2]
```diff
@@ -142,6 +142,23 @@ function SiteController($scope, $rootScope, $state, $stateParams, $mdToast,
     $scope.updateSite($scope.editing, dolphin ? dolphin.id : null);
   });
 
+  $scope.$on("gonevisDash.SiteTemplatesModalController:setTemplate", function (event, data) {
+    var templateConfig = data.data.config;
+
+    API.SiteSetTemplate.put({siteId: site}, {site_template_id: data.data.id},
+      function () {
+        $scope.siteTemplate = templateConfig;
+        $mdToast.showSimple("Site template updated");
```
Comment (Alir3z4) [inline app/site/site_controller.js]: End with a dot

File: app/site/site_controller.js [hunk 3]
```diff
@@ -142,6 +142,23 @@ function SiteController($scope, $rootScope, $state, $stateParams, $mdToast,
...
+      }, function () {
+        $mdToast.showSimple("Oh... Couldn't update site template");
```
Comment (Alir3z4) [inline app/site/site_controller.js]: What the about the actual error ?
Doesn't backend send an error ?

"Oh... Couldn't change the template."
if there's an error from backend, then show it.

Comment (amirm3hdi) [inline app/site/site_controller.js]: @Alir3z4 No, there's no form and we don't want to too much code,
If site template couldn't be updated there's a major error that backend will fail to show.

**Pull request merged**